### PR TITLE
Feature Request: support to clean expired entries

### PR DESCRIPTION
**Pull request by flisky**

### CODE REVIEW

**Comment by flisky** [comment]
Emm, there is a data race indeed, but maybe it's acceptable?

**Comment by flisky** [comment]
Ping, anyone?

If you don't like this, is 17e9a04 okay? I can split it out.

File: bigcache.go [hunk 1]
```diff
@@ -58,6 +59,14 @@ func newBigCache(config Config, clock clock) (*BigCache, error) {
 		cache.shards[i] = initNewShard(config, onRemove)
 	}
 
+	if config.CleanWindow >= time.Second {
```
Comment (janisz) [inline bigcache.go]: I'd prefer to check if this is ≥ 0
We need to document this behavior

File: config.go [hunk 2]
```diff
@@ -8,6 +8,8 @@ type Config struct {
 	Shards int
 	// Time after which entry can be evicted
 	LifeWindow time.Duration
+	// Time tick to clean up expired entries
```
Comment (janisz) [inline config.go]: Please describe more this feature and how to disable it.

**Comment by janisz** [comment]
@flisky I'm sorry for the delay. I've left some comments.

File: bigcache.go (hunk 1 above)
Comment (flisky) [inline bigcache.go]: Our timestamp is in unix epoch. Any tick <1 second is sort of wasting CPU.

Anyway, it's not intuitive. I'll change it.

File: config.go (hunk 2 above)
Comment (flisky) [inline config.go]: Non-native English speaker. Any suggestion is appreciated.

File: bigcache.go (hunk 1 above)
Comment (janisz) [inline bigcache.go]: > Our timestamp is in unix epoch. Any tick <1 second is sort of wasting CPU.

Good point. Please add this in godoc for CleanWindow field

File: config.go (hunk 2 above)
Comment (janisz) [inline config.go]: How about:

> // Interval between removing expired entries (clean up). If set to ≤ 0 then no action is performed. Setting to < 1 second is counterproductive — bigcache has a one second resolution.

Comment (flisky) [inline config.go]: Updated, thanks!

**Comment by janisz** [comment]
@flisky Thanks!

**Pull request merged**